python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -768,63 +775,174 @@ def contains_properly(self, other, align=True):
             A Series of boolean values indicating whether each point falls
             within the corresponding polygon in the input.
         """
-        if not contains_only_polygons(self):
-            raise TypeError(
-                "`.contains` can only be called with polygon series."
-            )
-
-        (lhs, rhs) = self.align(other) if align else (self, other)
-
-        # RHS conditioning:
-        mode = "POINTS"
-        # point in polygon
-        if contains_only_linestrings(rhs):
-            # condition for linestrings
-            mode = "LINESTRINGS"
-            geom = rhs.lines
-        elif contains_only_polygons(rhs) is True:
-            # polygon in polygon
-            mode = "POLYGONS"
-            geom = rhs.polygons
-        elif contains_only_multipoints(rhs) is True:
-            # mpoint in polygon
-            mode = "MULTIPOINTS"
-            geom = rhs.multipoints
+        if contains_only_points(self) and contains_only_points(other):
+            return cudf.Series(EqualsBinpred(self, other, align)())
         else:
-            # no conditioning is required
-            geom = rhs.points
-        xy_points = geom.xy
-        point_indices = geom.point_indices()
-        points = GeoSeries(GeoColumn._from_points_xy(xy_points._column)).points
-
-        # call pip on the three subtypes on the right:
-        point_result = contains_properly(
-            points.x,
-            points.y,
-            lhs.polygons.part_offset[:-1],
-            lhs.polygons.ring_offset[:-1],
-            lhs.polygons.x,
-            lhs.polygons.y,
-        )
-        if (
-            mode == "LINESTRINGS"
-            or mode == "POLYGONS"
-            or mode == "MULTIPOINTS"
-        ):
-            # process for completed linestrings, polygons, and multipoints.
-            # Not necessary for points.
-            result = cudf.DataFrame(
-                {"idx": point_indices, "pip": point_result}
-            )
-            # if the number of points in the polygon is equal to the number of
-            # points, then the requirements for `.contains_properly` are met
-            # for this geometry type.
-            df_result = (
-                result.groupby("idx").sum().sort_index()
-                == result.groupby("idx").count().sort_index()
+            return ContainsProperlyBinpred(self, other, align)()
+
+    def equals(self, other, align=True):
+        """Compute if a GeoSeries of features A is equal to a GeoSeries of
+        features B. Features are equal if their coordinates are equal.
```

Comment:
Are they equal if they have all the same vertices but in a different order? E.g. rotated by one index?

```suggestion
        features B. Features are equal if their vertices are equal.
```